src/hotspot/share/cds/dumpTimeClassInfo.hpp

```diff
@@ -40,32 +40,93 @@ class DumpTimeClassInfo: public CHeapObj<mtClass> {
   bool                         _excluded;
   bool                         _is_early_klass;
   bool                         _has_checked_exclusion;
-public:
-  struct DTLoaderConstraint {
+
+  static void maybe_increment_refcount(Symbol* s) {
+    if (s != NULL) {
+      s->increment_refcount();
+    }
+  }
+  static void maybe_decrement_refcount(Symbol* s) {
+    if (s != NULL) {
+      s->decrement_refcount();
+    }
```

Comment:
Can you put these two functions as inline functions in symbol.hpp.  We do this way too much in lots of code.